File: AspNetMvcUnleashed/0672329980_source/Chapter05/VB/MvcApplication1/MvcApplication1/Controllers/HomeController.vb
Public Class HomeController
    Inherits System.Web.Mvc.Controller

    Private _entities As New ProductsDBEntities()

    '
    ' GET: /Home/

    Function Index() As ActionResult
Dim results = From p In _entities.ProductSet _
      Where p.Price > 10.0 _
      Order By p.Name _
      Select p

        Return View(_entities.ProductSet.ToList())
    End Function

    '
    ' GET: /Home/Details/5

    Function Details(ByVal id As Integer) As ActionResult
        Dim result = (From p In _entities.ProductSet _
                      Where p.Id = id _
                      Select p).FirstOrDefault()

        Return View(result)
    End Function

    '
    ' GET: /Home/Create

    Function Create() As ActionResult
        Return View()
    End Function

    '
    ' POST: /Home/Create

    <AcceptVerbs(HttpVerbs.Post)> _
    Function Create(productToCreate As Product) As ActionResult
        Try
            _entities.AddToProductSet(productToCreate)
            _entities.SaveChanges()
            Return RedirectToAction("Index")
        Catch
            Return View()
        End Try
    End Function

    '
    ' GET: /Home/Edit/5

    Function Edit(ByVal id As Integer) As ActionResult
        Dim productToEdit = (From p In _entities.ProductSet _
                      Where p.Id = id _
                      Select p).FirstOrDefault()
        Return View(productToEdit)
    End Function

    '
    ' POST: /Home/Edit/5

    <AcceptVerbs(HttpVerbs.Post)> _
    Function Edit(productToEdit As Product) As ActionResult
        Try
            Dim originalProduct = (From p In _entities.ProductSet _
                      Where p.Id = productToEdit.Id _
                      Select p).FirstOrDefault()
            _entities.ApplyPropertyChanges(originalProduct.EntityKey.EntitySetName, productToEdit)
            _entities.SaveChanges()
            Return RedirectToAction("Index")
        Catch
            Return View()
        End Try
    End Function

    '
    ' GET: /Home/Delete/5

    Function Delete(ByVal id As Integer) As ActionResult
        Dim productToDelete = (From p In _entities.ProductSet _
                      Where p.Id = id _
                      Select p).FirstOrDefault()
        Return View(productToDelete)
    End Function

    '
    ' POST: /Home/Delete

    <AcceptVerbs(HttpVerbs.Post)> _
    Function Delete(productToDelete As Product) As ActionResult
        Try
            Dim originalProduct = (From p In _entities.ProductSet _
                      Where p.Id = productToDelete.Id _
                      Select p).FirstOrDefault()
            _entities.DeleteObject(originalProduct)
            _entities.SaveChanges()
            Return RedirectToAction("Index")
        Catch
            Return View()
        End Try
    End Function


End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter05/CS/MvcApplication1/MvcApplication1/Controllers/HomeController.cs
using System.Linq;
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class HomeController : Controller
    {
        private ProductsDBEntities _entities = new ProductsDBEntities();


        //
        // GET: /Home/

        public ActionResult Index()
        {
            return View(_entities.ProductSet.ToList());
        }

        //
        // GET: /Home/Details/5

        public ActionResult Details(int id)
        {
            var result = (from p in _entities.ProductSet 
                          where p.Id == id 
                          select p).FirstOrDefault();
            return View(result);
        }

        //
        // GET: /Home/Create

        public ActionResult Create()
        {
            return View();
        } 

        //
        // POST: /Home/Create

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create(Product productToCreate)
        {
            try
            {
                _entities.AddToProductSet(productToCreate);
                _entities.SaveChanges();

                return RedirectToAction("Index");
            }
            catch
            {
                return View();
            }
        }

        //
        // GET: /Home/Edit/5

        public ActionResult Edit(int id)
        {
            var productToEdit = (from p in _entities.ProductSet
                          where p.Id == id
                          select p).FirstOrDefault();
            return View(productToEdit);
        }

        //
        // POST: /Home/Edit/5

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Edit(Product productToEdit)
        {
            try
            {
                var originalProduct = (from p in _entities.ProductSet
                                     where p.Id == productToEdit.Id
                                     select p).FirstOrDefault();
                _entities.ApplyPropertyChanges(originalProduct.EntityKey.EntitySetName, productToEdit);
                _entities.SaveChanges();
                return RedirectToAction("Index");
            }
            catch
            {
                return View();
            }
        }


        //
        // GET: /Home/Delete/5

        public ActionResult Delete(int id)
        {
            var productToDelete = (from p in _entities.ProductSet
                                 where p.Id == id
                                 select p).FirstOrDefault();
            return View(productToDelete);
        }

        //
        // POST: /Home/Delete/5

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Delete(Product productToDelete)
        {
            try
            {
                var originalProduct = (from p in _entities.ProductSet
                                       where p.Id == productToDelete.Id
                                       select p).FirstOrDefault();

                _entities.DeleteObject(originalProduct);
                _entities.SaveChanges();
                return RedirectToAction("Index");
            }
            catch
            {
                return View();
            }
        }

    
    
    }
}
